zlib-rs/src/inflate.rs

```diff
@@ -862,12 +844,37 @@ impl State<'_> {
     }
 
     fn dispatch(&mut self) -> ReturnCode {
```

Comment:
just to be sure, let's add a comment around the top somewhere warning that early returns must restore the `mode`

Reply:
done